src/lib.rs

```diff
@@ -529,49 +560,157 @@ pub mod pallet {
 		///
 		/// # <weight>
 		#[transactional]
-		#[pallet::weight(T::WeightInfo::create_supersig())]
-		pub fn leave_supersig(origin: OriginFor<T>, supersig_id: T::AccountId) -> DispatchResult {
+		#[pallet::weight(T::WeightInfo::leave_supersig())]
+		pub fn leave_supersig(
+			origin: OriginFor<T>,
+			supersig_account: T::AccountId,
+		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
+			let supersig_id = Self::get_supersig_id_from_account(&supersig_account)?;
 
-			let supersig_index = Self::get_supersig_index_from_id(&supersig_id)?;
-			let supersig = Self::supersigs(supersig_index).unwrap();
-
-			if !supersig.is_user_in_supersig(&who) {
-				return Err(Error::<T>::NotMember.into())
+			if Self::members(supersig_id, &who) == Role::NotMember {
+				return Err(Error::<T>::NotMember.into());
 			}
 
-			Supersigs::<T>::try_mutate(supersig_index, |wrapped_supersig| {
-				if let Some(supersig) = wrapped_supersig {
-					if supersig.members.len() <= 1 {
-						return Err(())
-					}
-					supersig.members.retain(|memb| memb != &who);
-				}
+			// A supersig should at least have one member
+			TotalMembers::<T>::try_mutate(supersig_id, |nb| {
+				if *nb == 1 {
+					return Err(Error::<T>::InvalidNumberOfMembers);
+				};
+
+				*nb -= 1;
+
 				Ok(())
-			})
-			.map_err(|_| Error::<T>::CannotRemoveUsers)?;
-			Self::deposit_event(Event::<T>::SupersigLeaved(supersig_id, who));
+			})?;
+			// Note that the votes the user made stays in storage
```

Comment:
No funds are release when an user leave the supersig

Reply:
Please do it haha